playground/infrastructure/helper.py

```diff
@@ -84,47 +99,53 @@ def get_statuses(examples: [Example]):
 
 
 def get_tag(filepath):
-    """Parse file by filepath and find beam tag (after the licence part, before the import part)
-
-    If file contains tag, returns tag as a map.
-    If file contains tag but tag has incorrect format, raise error
-    If file doesn't contain tag, returns None
+    """
+    Parse file by filepath and find beam tag
 
     Args:
         filepath: path of the file
+
+    Returns:
+        If file contains tag, returns tag as a map.
+        If file doesn't contain tag, returns None
     """
+    add_to_yaml = False
+    yaml_string = ""
+
     with open(filepath) as parsed_file:
-        content = parsed_file.read()
-    index_of_licence_end = content.find(END_OF_LICENCE) + len(END_OF_LICENCE)
-    index_of_import_start = content.find(START_OF_IMPORT)
-    content = content[index_of_licence_end:index_of_import_start]
-
-    index_of_tag_start = content.find(BEAM_PLAYGROUND)
-    if index_of_tag_start < 0:
-        return None
-    index_of_from_start = content.find(START_OF_FROM)
-    if index_of_from_start > 0:
-        content = content[index_of_tag_start:index_of_from_start]
-    else:
-        content = content[index_of_tag_start:]
-    yaml_tag = content.replace("//", "").replace("#", "")
-    try:
-        object_meta = yaml.load(yaml_tag, Loader=yaml.SafeLoader)
-        return object_meta[BEAM_PLAYGROUND]
-    except Exception as exp:
-        print(exp)  ## todo add logErr
-        raise ValueError("found tag is not correct: " + exp.__str__())
+        lines = parsed_file.readlines()
+
+    for line in lines:
+        line = line.replace("// ", "").replace("# ", "")
+        if add_to_yaml is False:
+            if line == BEAM_PLAYGROUND_TITLE:
+                add_to_yaml = True
+                yaml_string += line
+        else:
+            yaml_with_new_string = yaml_string + line
+            try:
+                yaml.load(yaml_with_new_string, Loader=yaml.SafeLoader)
+                yaml_string += line
+            except Exception:
+                break
+
+    if add_to_yaml:
+        tag_object = yaml.load(yaml_string, Loader=yaml.SafeLoader)
+        return tag_object[BEAM_PLAYGROUND]
```

Comment:
Shouldn't here be `BEAM_PLAYGROUND_TITLE` instead of `BEAM_PLAYGROUND`?

Reply:
No, we use `BEAM_PLAYGROUND_TITLE` like a exact string which we should find in the file.
`BEAM_PLAYGROUND` is used as a key for `yaml`.

Reply:
Got it, thanks!